src/preprocessing/shapes/shapes.jl

```diff
@@ -0,0 +1,13 @@
+abstract type Shapes{NDIMS} end
```

Comment:
This is a bit confusing. `ComplexShape` is no `Shapes`? And `RectangularShape` isn't either? I think you should find a different name for this abstract type.

Reply:
how about `AbstractGeometry`?

Reply:
I would prefer to rename several things in this pr to geometry since this seems to me a more common name for what this pr is doing. While the concept of a shape is more limited to our own nomenclature.

Reply:
> how about AbstractGeometry?
?